TileEngine/Isometric Utils.cpp

```diff
@@ -912,6 +917,14 @@ INT16 sDeltaScreenX, sDeltaScreenY;
 
 }
 
+INT16 DirectionToVectorX[] = {0, 1, 1, 1, 0, -1, -1, -1, 0};
+INT16 DirectionToVectorY[] = {-1, -1, 0, 1, 1, 1, 0, -1, 0};
+void ConvertDirectionToVectorInXY( UINT8 ubDirection, INT16* sXDir, INT16* sYDir )
+{
+	*sXDir = DirectionToVectorX[ubDirection];
+	*sYDir = DirectionToVectorY[ubDirection];
+}

```

Comment:
I am aware of the out of bounds if inputting the wrong data, but whos job is to make sure that out of bound is not added and how? (In other places they do modulo which is imo wrong in this occasion)

Is std pair used somewhere else? I rather would like to introduce a proper position_2d thats used everywhere, but im scared and rather just copy existing style

Reply:
not that I know of. the `position_2d` type-safety is a great idea. creating a `direction` type would also nip this one right in the bud.

maybe just put an assert for now?

Reply:
assert is good enough for now. we really are missing basic structures everywhere. so many code clones because of missing shared code.. its unbelievable.

Reply:
std::pair is also used alot, but "Isometric Utils.cpp" is really more about C style,  so don't want to use it here